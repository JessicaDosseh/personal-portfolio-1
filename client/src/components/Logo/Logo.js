import React from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const LogoDiv = styled.div`
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  height: 130px;
  width: 130px;
  background: #845ae4;
  display: flex;
  justify-content: center;
  align-items: center;

  ${props =>
    !props.darkmode &&
    css`
      -webkit-box-shadow: 8px 10px 37px -14px rgba(0, 0, 0, 0.42);
      -moz-box-shadow: 8px 10px 37px -14px rgba(0, 0, 0, 0.42);
      box-shadow: 8px 10px 37px -14px rgba(0, 0, 0, 0.42);
    `}

  ${props =>
    props.darkmode &&
    css`
      background: black;
    `} /*  #181818   */

  div {
    border-radius: 50%;
    -moz-border-radius: 50%;
    -webkit-border-radius: 50%;
    height: 120px;
    width: 120px;
    background: #845ae4;
    display: flex;
    justify-content: center;
    align-items: center;

    ${props =>
      props.darkmode &&
      css`
        background: black;
      `}

    p {
      font-family: "Work Sans", sans-serif;
      font-size: 12rem;
      font-weight: 700;
      color: white;
    }
  }
`;

const Logo = props => {
  return (
    <Link to="/">
      <LogoDiv darkmode={props.darkmode}>
        <div>
          <p>N</p>
        </div>
      </LogoDiv>
    </Link>
  );
};

Logo.propTypes = {
  darkmode: PropTypes.bool.isRequired
};

export default Logo;
