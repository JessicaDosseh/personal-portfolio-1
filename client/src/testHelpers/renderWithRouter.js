import React from "react";
import { Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
