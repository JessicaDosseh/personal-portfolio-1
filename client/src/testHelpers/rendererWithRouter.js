import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
