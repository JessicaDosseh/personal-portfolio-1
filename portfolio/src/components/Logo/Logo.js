import React from "react";
import styled, { css } from "styled-components";

const LogoDiv = styled.div`
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  height: 130px;
  width: 130px;
  background: #845ae4;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-shadow: -1px 4px 34px 0px rgba(0, 0, 0, 0.18);
  -moz-box-shadow: -1px 4px 34px 0px rgba(0, 0, 0, 0.18);
  box-shadow: -1px 4px 34px 0px rgba(0, 0, 0, 0.18);

  ${props =>
    props.darkmode &&
    css`
      background: black;
    `} /*  #181818   */

  div {
    border-radius: 50%;
    -moz-border-radius: 50%;
    -webkit-border-radius: 50%;
    height: 120px;
    width: 120px;
    background: #845ae4;
    display: flex;
    justify-content: center;
    align-items: center;

    ${props =>
      props.darkmode &&
      css`
        background: black;
      `}

    p {
      font-family: "Work Sans", sans-serif;
      font-size: 12rem;
      font-weight: 700;
      color: white;
    }
  }
`;

const Logo = props => {
  return (
    <LogoDiv darkmode={props.darkmode}>
      <div darkmode={props.darkmode}>
        <p>N</p>
      </div>
    </LogoDiv>
  );
};

export default Logo;
