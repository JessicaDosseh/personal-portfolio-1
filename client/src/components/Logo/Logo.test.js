import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import Logo from "./Logo";

describe("<Logo />", () => {
  it("should render the Logo component without crashing", () => {
    const helpers = renderWithRouter(<Logo darkmode={false} />);
  });

  it("should match the snapshot of Logo", () => {
    const tree = rendererWithRouter(<Logo darkmode={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
