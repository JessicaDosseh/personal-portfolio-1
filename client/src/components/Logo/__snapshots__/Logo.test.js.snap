// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match the snapshot of Logo 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <div
    className="sc-AykKC fhSaEQ"
  >
    <div>
      <p>
        N
      </p>
    </div>
  </div>
</a>
`;
